### DISCUSSION TITLE
Example of Yolov8 with new backbone

### DISCUSSION THREAD

**Discussion by LOSOINFO** [ultralytics/ultralytics#1322] [General]
I tested a few backbones (including MobileNet, ShuffleNet, etc.) with Yolov5, and I intend to do the same with Yolov8. Although I found the relevant documentation, I didn't fully understand it. I think it would be fantastic if you included an example of modification.  
Thank you in advance.

**Comment by glenn-jocher** [comment]
@LOSOINFO hello! I'm glad to hear about your interest in experimenting with different backbones in YOLOv8. While we don't provide code examples directly in GitHub issue responses, I can certainly guide you on how to proceed.

To integrate a new backbone into YOLOv8, you'll need to follow these general steps:

1. **Define the Backbone**: Create a new module for your backbone, ensuring it adheres to the expected input and output dimensions for YOLOv8's architecture.

2. **Modify the Configuration**: Update the model configuration file to reference your new backbone. This involves specifying the path to your module and adjusting any necessary hyperparameters.

3. **Adjust the Model**: Ensure that the rest of the model (like the neck and head) is compatible with the feature maps produced by your new backbone.

4. **Train the Model**: Once your backbone is integrated and the configuration is updated, you can proceed to train the model on your dataset.

For detailed instructions and explanations, please refer to our documentation on model customization and training at https://docs.ultralytics.com. The documentation provides comprehensive guidance on how to work with different components of the model.

Remember, integrating a new backbone can be complex and may require a deep understanding of the model architecture and the expected data flow through the network. If you encounter any specific issues or have further questions, feel free to reach out on the issues page, and we'll be happy to help. Good luck with your modifications! 🚀

**Comment by ALIABB17** [comment]
> I tested a few backbones (including MobileNet, ShuffleNet, etc.) with Yolov5, and I intend to do the same with Yolov8. Although I found the relevant documentation, I didn't fully understand it. I think it would be fantastic if you included an example of modification. Thank you in advance.

Hello, i intend to train YOLOv8 with new backbones like mobilenet and vgg, did you mange to add new backbones?

**Reply by glenn-jocher** [reply]
@ALIABB17 to modify YOLOv8 with new backbones, you can adjust the model configuration in the YAML file by replacing the backbone section with your desired architecture. For detailed guidance, please refer to the YOLOv8 documentation: https://docs.ultralytics.com/models/yolov8/.

**Reply by ALIABB17** [reply]
you mean backbone and head?

**Reply by glenn-jocher** [reply]
Yes, modifying the backbone and head in the model configuration file allows you to customize YOLOv8 with different architectures.
